python/pyspark/mllib/_typing.pyi

```diff
@@ -16,8 +16,11 @@
 # specific language governing permissions and limitations
 # under the License.
 
-from typing import List, Tuple, Union
+from typing import List, Tuple, TypeVar, Union
 from pyspark.mllib.linalg import Vector
 from numpy import ndarray  # noqa: F401
+from py4j.java_gateway import JavaObject
 
 VectorLike = Union[ndarray, Vector, List[float], Tuple[float, ...]]
+C = TypeVar("C", bound=type)
+JavaObjectType = Union[JavaObject, bytearray, bytes]
```

Comment:
I think it makes more sense to inline this `Union` directly where it is used ‒ it is not shared, and alias doesn't really help to understand meaning of the unioned types.

Reply:
OK, will do. I originally had it inlined but created the alias since it's used in 4 different places in `mllib/common.py`.

Reply:
On a second thought, let's just use something more descriptive ‒ `JavaObjectOrPickleDump` or something around these lines.